### ISSUE TITLE
Can't add local SPM packages using Dependencies.swift

### ISSUE THREAD

**Issue by zeevvax** [tuist/tuist#4234]
**Describe the bug**
I have few modules with SPM packages as part of my project, 
Local packageB has a dependency on local packageA.
Added PackageA to Dependencies.swift with
.local(path: "../packageA")

In PackageB project.swift have 
let dependencies: [TargetDependency] = [
    .external(name: "packageA"),
    .external(name: "BetterCodable")
]

**To Reproduce**

1. Run >> tuist fetch
2. Run >>tuist generate

**Expected behavior**
The project was generated with no errors 

**Actual result**
Resolved cache profile 'Development' from Tuist's defaults
`packageA` is not a valid configured external dependency

note: The Package that was created under Tuist/Dependencies/SwiftPackageManager/ 
showing correctly the package information with the local dependencies 

**Screenshots**
Here are my local package with the info I described above 
<img width="821" alt="Screen Shot 2022-03-11 at 5 52 27 PM" src="https://user-images.githubusercontent.com/1322662/157999265-66c170c0-77d3-4b22-970d-dc431f208729.png">
<img width="606" alt="Screen Shot 2022-03-11 at 5 53 07 PM" src="https://user-images.githubusercontent.com/1322662/157999266-dcd65ef2-297a-41b9-868f-e5510fa61665.png">
<img width="877" alt="Screen Shot 2022-03-11 at 5 53 47 PM" src="https://user-images.githubusercontent.com/1322662/157999268-451047c0-7403-4c07-8f0b-f9660aa076b3.png">

**Comment by danyf90** [comment]
The value passed to `.external` must be a product name, which is what you can import from packages. Your package has not products, and hence that error is expected.
Could you please clarify what are you trying to do with that package?

**Comment by zeevvax** [comment]
Thanks, Dany, I'm trying to have multiple independent development modules, and they have some dependencies between them. I can't attached files in my reply so below is the TILogging package, and the dependencies P`ackage.resolved` that was created under `Tuist/SwiftPackageManager `after I run `tuist fetch` looks like the fetch didn't resolve TILogging and my other local package to a package (but did resolve their dependencies). 
`// swift-tools-version:5.3
// The swift-tools-version declares the minimum version of Swift required to build this package.
import PackageDescription

let package = Package(
    name: "TILogging",
    platforms: [
        .iOS(.v14)
    ],
    products: [
        .library(
            name: "TILogging",
            targets: ["TILogging"]
        )
    ],
    dependencies: [
        .package(
            name: "CocoaLumberjack",
            url: "https://github.com/CocoaLumberjack/CocoaLumberjack.git",
            from: "3.7.2"
        ),
        .package(
            name: "Nimble",
            url: "https://github.com/Quick/Nimble.git",
            from: "9.0.1"
        ),
        .package(
            name: "Quick",
            url: "https://github.com/Quick/Quick.git",
            from: "3.1.2"
        ),
        .package(
            name: "ZIPFoundation",
            url: "https://github.com/weichsel/ZIPFoundation.git",
            from: "0.9.11"
        )
    ],
    targets: [
        .target(
            name: "TILogging",
            dependencies: [
                "CocoaLumberjack",
                "ZIPFoundation"
            ]
        ),
        .testTarget(
            name: "TILoggingTests",
            dependencies: [
                .target(name: "TILogging"),
                .product(name: "Quick", package: "Quick"),
                .product(name: "Nimble", package: "Nimble")
            ]
        )
    ]
)
`

`{
  "object": {
    "pins": [
      {
        "package": "BetterCodable",
        "repositoryURL": "https://github.com/marksands/BetterCodable.git",
        "state": {
          "branch": null,
          "revision": "61153170668db7a46a20a87e35e70f80b24d4eb5",
          "version": "0.4.0"
        }
      },
      {
        "package": "CocoaLumberjack",
        "repositoryURL": "https://github.com/CocoaLumberjack/CocoaLumberjack.git",
        "state": {
          "branch": null,
          "revision": "80ada1f753b0d53d9b57c465936a7c4169375002",
          "version": "3.7.4"
        }
      },
      {
        "package": "Commander",
        "repositoryURL": "https://github.com/kylef/Commander.git",
        "state": {
          "branch": null,
          "revision": "4a1f2fb82fb6cef613c4a25d2e38f702e4d812c2",
          "version": "0.9.2"
        }
      },
      {
        "package": "rswift",
        "repositoryURL": "https://github.com/mac-cain13/R.swift.git",
        "state": {
          "branch": null,
          "revision": "994c2a92435b831d3dd184fc98936ff38e874106",
          "version": "6.1.0"
        }
      },
      {
        "package": "Spectre",
        "repositoryURL": "https://github.com/kylef/Spectre.git",
        "state": {
          "branch": null,
          "revision": "26cc5e9ae0947092c7139ef7ba612e34646086c7",
          "version": "0.10.1"
        }
      },
      {
        "package": "swift-log",
        "repositoryURL": "https://github.com/apple/swift-log.git",
        "state": {
          "branch": null,
          "revision": "5d66f7ba25daf4f94100e7022febf3c75e37a6c7",
          "version": "1.4.2"
        }
      },
      {
        "package": "XcodeEdit",
        "repositoryURL": "https://github.com/tomlokhorst/XcodeEdit",
        "state": {
          "branch": null,
          "revision": "99547c5af5850155b52c43b716ba1b094b02a3b2",
          "version": "2.8.0"
        }
      },
      {
        "package": "ZIPFoundation",
        "repositoryURL": "https://github.com/weichsel/ZIPFoundation.git",
        "state": {
          "branch": null,
          "revision": "7254c74b49cec2cb81520523ba993c671f71b066",
          "version": "0.9.14"
        }
      }
    ]
  },
  "version": 1
}
`

**Comment by danyf90** [comment]
Hi @zeevvax 👋 
It's very hard to follow it in the github message, could you please create a sample project to reproduce the issue?

**Comment by zeevvax** [comment]
Hi Daniele,

Sorry it took me so long when I tried to create a sample project as per
your request I realized I should do a Tuist upgrade for other Packages.
Saying that, attached is a simple Privacy module, I include two projects in
the Archive.
Privacy - has the PrivacyTestApp depends on a Privacy target in the
Project.swift file, and the external dependency on OTPublishersHeadlessSDK is
defined in the project file + dependency file. This project can be created
with tuist generate compile and run. The issue with this is that is that I
don't see Privacy as a SPM dependency (the eventual intended usage) when
loading the workspace
PrivacyLocal - has the PrivacyTestApp depends on a local Privacy package
which define a dependency on Privacy in the dependency file. This one fail
to tuist generate with this error.
"Swift/x86_64-apple-macos.swiftinterface:13515: Fatal error: Duplicate
values for key: '/Users/zeevvax/Downloads/PrivacyLocal'

I can also build a sample with two local Packages that one is dependent on
the other, but thought that this sample shows a similar issue.

Thanks,
Zeev
 Archive.zip
<https://drive.google.com/file/d/1ZV12sB4MBjg2U6X5SGSRSAfX4oxlldHk/view?usp=drive_web>

Ze'ev Vax (he/him)
Director of Engineering, iOS
M: 415.412.9952
***@***.*** ***@***.***>*
www.tunein.com
